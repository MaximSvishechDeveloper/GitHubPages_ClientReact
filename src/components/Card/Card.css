.github-link {
  font-size: 20px;
  color: black;
  padding: 10px 20px;
  background-color: transparent;
  border: 1px solid black;
  cursor: pointer;
  text-decoration: none;
}

#light .card {
  background-color: #0ccac4;
}

#dark .card {
  background-color: black;
}

#dark .dev-name {
  color: white;
}

#dark .github-link {
  color: white;
  background-color: gray;
}
