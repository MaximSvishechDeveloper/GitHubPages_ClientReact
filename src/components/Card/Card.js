import React from "react";
import "./Card.css";

const Card = ({ name, id, page, image }) => {
  const handleButtonClick = () => {
    window.location.href = page;
  };

  return (
    <div className="card tc dib br3 pa3 ma2 grow bw2 shadow-5">
      <img
        alt="profiles"
        src={image}
        style={{ width: "400px", height: "400px" }}
      />
      <div>
        <h2 className="dev-name">{name}</h2>
        <button className="github-link" onClick={handleButtonClick}>
          GitHub Page
        </button>
      </div>
    </div>
  );
};

export default Card;
