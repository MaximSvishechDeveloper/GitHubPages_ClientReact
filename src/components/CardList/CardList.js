import React from "react";
import Card from "../Card/Card";
import { v4 as uuidv4 } from "uuid";

const CardList = ({ profiles }) => {
  return (
    <div>
      {profiles.map((user, i) => {
        return (
          <Card
            key={uuidv4()}
            id={profiles[i].id}
            name={profiles[i].login}
            page={profiles[i].html_url}
            image={profiles[i].avatar_url}
          />
        );
      })}
    </div>
  );
};

export default CardList;
