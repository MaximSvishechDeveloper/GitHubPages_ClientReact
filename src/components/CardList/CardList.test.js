import React from "react";
import { shallow } from "enzyme";
import CardList from "./CardList";
import Card from "../Card/Card";

describe("CardList component", () => {
  it("should render Card components with correct props", () => {
    const mockProfiles = [
      {
        id: 1,
        login: "john_doe",
        html_url: "https://example.com/john_doe",
        avatar_url: "https://example.com/avatar.jpg",
      },
      {
        id: 2,
        login: "jane_doe",
        html_url: "https://example.com/jane_doe",
        avatar_url: "https://example.com/avatar2.jpg",
      },
    ];

    const wrapper = shallow(<CardList profiles={mockProfiles} />);

    expect(wrapper.find(Card)).toHaveLength(mockProfiles.length);

    mockProfiles.forEach((profile, index) => {
      const cardProps = wrapper.find(Card).at(index).props();
      expect(cardProps.id).toEqual(profile.id);
      expect(cardProps.name).toEqual(profile.login);
      expect(cardProps.page).toEqual(profile.html_url);
      expect(cardProps.image).toEqual(profile.avatar_url);
    });
  });
});
