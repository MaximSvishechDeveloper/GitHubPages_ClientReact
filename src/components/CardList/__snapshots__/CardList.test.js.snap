// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render cardList 1`] = `
<div>
  <Card
    id={1}
    key="1f711b21-2b9b-4bc3-bbac-d2a999c9251f"
  />
</div>
`;
